core/src/main/java/io/grpc/internal/Stream.java

```diff
@@ -77,4 +79,10 @@
    * result in excessive buffering within the transport.
    */
   boolean isReady();
+
+  /**
+   * Sets the default message encoder for messages on this stream.
+   * @param c the compressor / decompressor
+   */
+  void setDefaultCodec(Codec c);
```

Comment:
I think this is the main place where we need a bit of design.  How do we configure the system to use codecs ... and what use cases do we support, etc.